src/gt4py/next/ffront/type_info.py

```diff
@@ -197,7 +197,9 @@ def _as_field(dtype: ts.TypeSpec, path: tuple[int, ...]) -> ts.FieldType:
             # TODO: we want some generic field type here, but our type system does not support it yet.
             return ts.FieldType(dims=[common.Dimension("...")], dtype=dtype)
 
-    return type_info.apply_to_primitive_constituents(param, _as_field, with_path_arg=True)
+    res = type_info.apply_to_primitive_constituents(param, _as_field, with_path_arg=True)
+    assert isinstance(res, (ts.FieldType, ts.TupleType))
```

Comment:
Why is the runtime check needed now? There are no changes in this function type hints..

Reply:
`apply_to_primitive_constituents` was not typed, but the return is not precise enough for how it's used here